### DISCUSSION TITLE
Python: locate filenames pointing to an archive

### DISCUSSION THREAD

**Discussion by Sim4n6** [github/codeql#11439] [Q&A] [answered]
Hi,  I would like to identify a source that points to an archive. 
For instance:
```python
filename1 = "/home/user/archive.zip"

from pathlib import Path
filename2 = Path("/home/user/archive.zip")
```

I have tried this query as a starting point:

```codeql
import semmle.python.Concepts

from DataFlow::Node source
where source = any(FileSystemAccess fa).getAPathArgument()
select source
```

But it seems the query would identify "data-flow nodes that perform a file system access" and not something "fixed".

Please any indication or help is a welcome to locate `filename1` and `filename2`.

**Comment by Sim4n6** [comment]
I went with the `source` as an `Str`. 

Thank you

**Reply by Sim4n6** [reply]
But it does not really answer my question. I mean if there is any different way to proceed, please feel free to suggest one.

**Comment by alexet** [comment, marked as answer]
If you want to find all string literals ending in ".zip" the following would work.
```
from StrConst s, DataFlow::Node n
where 
  n.asExpr() = s and
  s.getText().matches("%.zip")
```

Then if you used taint-flow you would see `filename2` was also tainted and therefore get similar results to if filename2 was the source.

**Reply by Sim4n6** [reply]
thx, the matches and getText that what I was looking for, again thanks
